### DISCUSSION TITLE
Why not using ThreadPool in WalkParallel::visit

### DISCUSSION THREAD

**Discussion by aiyanbo** [BurntSushi/ripgrep#2145] [Q&A] [answered]
https://github.com/BurntSushi/ripgrep/blob/0b36942f680bfa9ae88a564f2636aa8286470073/crates/ignore/src/walk.rs#L1285-L1306

**Comment by BurntSushi** [comment, marked as answer]
I don't know what the heck you're asking. Please add more detail to your question.

Also, consider that instead of asking "why not," you instead make an argument why I _should_ use something different. Then I can be more precise in telling you why I (probably) disagree.

I'm also moving this to a Discussion, since this seems like Q&A.
